### ISSUE TITLE
🐛 [firebase_analytics] 6.0.0 is not compatible with FirebaseCore after migration

### ISSUE THREAD

**Issue by AAverin** [FirebaseExtended/flutterfire#3479]
## Bug report

**Describe the bug**

After following https://firebase.flutter.dev/docs/migration I get the error when building for iOS

```
[!] CocoaPods could not find compatible versions for pod "Firebase/CoreOnly":
  In snapshot (Podfile.lock):
    Firebase/CoreOnly (= 6.30.0)

  In Podfile:
    firebase_analytics (from `.symlinks/plugins/firebase_analytics/ios`) was resolved to 6.0.0, which depends on
      Firebase/CoreOnly (~> 6.26.0)
```

### Steps to reproduce

User firebase_analytics 6.0.0
Follow https://firebase.flutter.dev/docs/migration
Try to build for iOS

### Expected behavior

Build passes correctly

---

## Additional context

Might be analytics also requires some version updates that are not yet published?

---

### Flutter doctor

Run `flutter doctor` and paste the output below:

<details><summary>Click To Expand</summary>

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 1.20.3, on Mac OS X 10.15.3 19D76, locale en-GB)
 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
[✓] Xcode - develop for iOS and macOS (Xcode 11.4.1)
[✓] Android Studio (version 3.5)
[✓] VS Code (version 1.48.2)
[✓] Connected device (2 available)
```

</details>

---

### Flutter dependencies

Run `flutter pub deps -- --style=compact` and paste the output below:

<details><summary>Click To Expand</summary>

```
Dart SDK 2.9.2
Flutter SDK 1.20.3
photo_hound 1.0.0+9

dependencies:
- build_resolvers 1.3.11 [analyzer build crypto graphs logging path package_config pub_semver]
- carousel_slider 2.2.1 [flutter]
- connectivity 0.4.9+2 [flutter meta connectivity_platform_interface connectivity_macos connectivity_for_web]
- cupertino_icons 0.1.3
- firebase_analytics 6.0.0 [meta flutter firebase_core firebase_analytics_web firebase_analytics_platform_interface]
- firebase_core 0.5.0 [firebase_core_platform_interface flutter quiver meta firebase_core_web]
- firebase_crashlytics 0.2.0-dev.5 [flutter stack_trace firebase_core firebase_core_platform_interface firebase_crashlytics_platform_interface]
- flt_worker 0.1.0 [flutter]
- flutter 0.0.0 [characters collection meta typed_data vector_math sky_engine]
- flutter_advanced_networkimage 0.8.0 [flutter http path_provider path flutter_svg]
- flutter_archive 1.0.2 [flutter]
- flutter_dotenv 2.1.0 [flutter meta]
- flutter_downloader 1.5.0 [flutter]
- flutter_hooks 0.13.2 [flutter]
- flutter_html 1.0.2 [html csslib css_colors video_player chewie webview_flutter chewie_audio flutter_svg flutter]
- flutter_redux 0.6.0 [meta redux flutter]
- flutter_secure_storage 3.3.3 [meta flutter]
- font_awesome_flutter 8.8.1 [flutter]
- get_it 4.0.4 [async meta]
- hive 1.4.4 [meta crypto]
- hive_flutter 0.3.1 [flutter hive path_provider path]
- http 0.12.2 [http_parser path pedantic]
- injectable 1.0.4 [get_it]
- json_serializable 3.4.0 [analyzer build build_config json_annotation meta path source_gen]
- logger 0.9.2
- mapbox_gl 0.8.0 [flutter mapbox_gl_platform_interface mapbox_gl_web]
- maps_launcher 1.2.2+1 [flutter flutter_web_plugins url_launcher]
- oauth1 1.0.5 [crypto http]
- path_provider 1.6.14 [flutter path_provider_platform_interface path_provider_macos path_provider_linux]
- permission_handler 5.0.1+1 [flutter meta permission_handler_platform_interface]
- photo_view 0.10.2 [flutter]
- provider 4.3.2+1 [flutter nested collection]
- redux 4.0.0+3
- redux_logging 0.4.0 [redux logging]
- redux_thunk 0.3.0 [redux]
- url_launcher 5.5.0 [flutter url_launcher_platform_interface url_launcher_web url_launcher_linux url_launcher_macos]

dev dependencies:
- build_runner 1.10.0 [args async build build_config build_daemon build_resolvers build_runner_core code_builder collection crypto dart_style glob graphs http_multi_server io js logging meta mime path pedantic pool pub_semver pubspec_parse shelf shelf_web_socket stack_trace stream_transform timing watcher web_socket_channel yaml]
- flutter_launcher_icons 0.7.5 [image args yaml]
- flutter_test 0.0.0 [flutter test_api path fake_async clock stack_trace vector_math async boolean_selector characters charcode collection matcher meta source_span stream_channel string_scanner term_glyph typed_data]
- hive_generator 0.7.1 [build source_gen hive analyzer dartx]
- injectable_generator 1.0.4 [build source_gen path glob analyzer injectable collection]

transitive dependencies:
- _fe_analyzer_shared 6.0.0 [meta]
- analyzer 0.39.14 [_fe_analyzer_shared args charcode cli_util collection convert crypto glob html meta package_config path pub_semver source_span watcher yaml]
- archive 2.0.13 [crypto args path]
- args 1.6.0
- async 2.4.2 [collection]
- boolean_selector 2.0.0 [source_span string_scanner]
- build 1.3.0 [analyzer async convert crypto logging meta path glob]
- build_config 0.4.2 [checked_yaml json_annotation meta path pubspec_parse yaml]
- build_daemon 2.1.4 [built_collection built_value http_multi_server logging pedantic path pool shelf shelf_web_socket stream_transform watcher web_socket_channel]
- build_runner_core 5.2.0 [async build build_config build_resolvers collection convert crypto glob graphs json_annotation logging meta path package_config pedantic pool timing watcher yaml]
- built_collection 4.3.2 [collection quiver]
- built_value 7.1.0 [built_collection collection fixnum quiver]
- characters 1.0.0
- charcode 1.1.3
- checked_yaml 1.0.2 [json_annotation source_span yaml]
- chewie 0.9.10 [open_iconic_flutter video_player wakelock flutter]
- chewie_audio 1.0.0+1 [open_iconic_flutter video_player screen flutter]
- cli_util 0.1.4 [path]
- clock 1.0.1 [meta]
- code_builder 3.4.1 [built_collection built_value collection matcher meta]
- collection 1.14.13
- connectivity_for_web 0.3.1+2 [connectivity_platform_interface flutter_web_plugins flutter]
- connectivity_macos 0.1.0+4 [flutter]
- connectivity_platform_interface 1.0.6 [flutter meta plugin_platform_interface]
- convert 2.1.1 [charcode typed_data]
- crypto 2.1.5 [collection convert typed_data]
- css_colors 1.0.2 [flutter]
- csslib 0.16.2 [source_span]
- dart_style 1.3.6 [analyzer args path source_span]
- dartx 0.5.0 [collection path crypto characters time meta]
- fake_async 1.1.0 [clock collection]
- file 5.2.1 [intl meta path]
- firebase 7.3.0 [http http_parser js]
- firebase_analytics_platform_interface 1.0.3 [flutter meta]
- firebase_analytics_web 0.1.1 [flutter flutter_web_plugins firebase firebase_analytics_platform_interface meta]
- firebase_core_platform_interface 2.0.0 [flutter meta plugin_platform_interface quiver]
- firebase_core_web 0.2.0 [firebase firebase_core_platform_interface flutter flutter_web_plugins meta js]
- firebase_crashlytics_platform_interface 1.0.0-dev.2 [flutter meta collection firebase_core plugin_platform_interface]
- fixnum 0.10.11
- flutter_svg 0.18.1 [path_drawing xml vector_math meta flutter]
- flutter_web_plugins 0.0.0 [flutter characters collection meta typed_data vector_math]
- glob 1.2.0 [async collection node_io path pedantic string_scanner]
- graphs 0.2.0
- html 0.14.0+3 [csslib source_span]
- http_multi_server 2.2.0 [async]
- http_parser 3.1.4 [charcode collection source_span string_scanner typed_data]
- image 2.1.14 [archive xml]
- intl 0.16.1 [path]
- io 0.3.4 [charcode meta path string_scanner]
- js 0.6.2
- json_annotation 3.0.1
- logging 0.11.4
- mapbox_gl_dart 0.1.5 [js meta]
- mapbox_gl_platform_interface 0.8.0 [flutter meta]
- mapbox_gl_web 0.8.0 [flutter flutter_web_plugins meta mapbox_gl_platform_interface mapbox_gl_dart image]
- matcher 0.12.8 [stack_trace]
- meta 1.1.8
- mime 0.9.7
- nested 0.0.4 [flutter]
- node_interop 1.1.1 [js]
- node_io 1.1.1 [node_interop path]
- open_iconic_flutter 0.3.0 [flutter]
- package_config 1.9.3 [path charcode]
- path 1.7.0
- path_drawing 0.4.1+1 [vector_math meta path_parsing flutter]
- path_parsing 0.1.4 [vector_math meta]
- path_provider_linux 0.0.1+2 [path xdg_directories path_provider_platform_interface flutter]
- path_provider_macos 0.0.4+3 [flutter]
- path_provider_platform_interface 1.0.3 [flutter meta platform plugin_platform_interface]
- pedantic 1.9.0
- permission_handler_platform_interface 2.0.1 [flutter meta plugin_platform_interface]
- petitparser 3.0.4 [meta]
- platform 2.2.1
- platform_detect 1.4.0 [meta pub_semver]
- plugin_platform_interface 1.0.2 [meta]
- pool 1.4.0 [async stack_trace]
- process 3.0.13 [file intl meta path platform]
- pub_semver 1.4.4 [collection]
- pubspec_parse 0.1.5 [checked_yaml json_annotation pub_semver yaml]
- quiver 2.1.3 [matcher meta]
- screen 0.0.5 [flutter]
- shelf 0.7.9 [async collection http_parser path stack_trace stream_channel]
- shelf_web_socket 0.2.3 [shelf web_socket_channel stream_channel]
- sky_engine 0.0.99
- source_gen 0.9.6 [analyzer async build dart_style glob meta path pedantic source_span]
- source_span 1.7.0 [charcode collection meta path term_glyph]
- stack_trace 1.9.5 [path]
- stream_channel 2.0.0 [async]
- stream_transform 1.2.0
- string_scanner 1.0.5 [charcode meta source_span]
- term_glyph 1.1.0
- test_api 0.2.17 [async boolean_selector collection meta path source_span stack_trace stream_channel string_scanner term_glyph matcher]
- time 1.3.0
- timing 0.1.1+2 [json_annotation]
- typed_data 1.2.0 [collection]
- url_launcher_linux 0.0.1+1 [flutter]
- url_launcher_macos 0.0.1+7 [flutter]
- url_launcher_platform_interface 1.0.7 [flutter meta plugin_platform_interface]
- url_launcher_web 0.1.2+1 [url_launcher_platform_interface platform_detect flutter flutter_web_plugins meta]
- vector_math 2.0.8
- video_player 0.10.12 [meta video_player_platform_interface video_player_web flutter]
- video_player_platform_interface 2.1.0 [flutter meta]
- video_player_web 0.1.3+2 [flutter flutter_web_plugins meta video_player_platform_interface]
- wakelock 0.1.4+2 [flutter]
- watcher 0.9.7+15 [async path pedantic]
- web_socket_channel 1.1.0 [async crypto stream_channel]
- webview_flutter 0.3.22+1 [flutter]
- xdg_directories 0.1.0 [path process flutter]
- xml 4.2.0 [collection convert meta petitparser]
- yaml 2.2.1 [charcode collection string_scanner source_span]

```

</details>

---

**Comment by darshankawar** [comment]
Hi @AAverin,
I added said dependency of `firebase_analytics` along with updated version of `firebase_core` and was able to build app on iOS device properly. I am on latest dev.

```
Running pod install...
Running Xcode build...
Xcode build done.                                           126.4s
Installing and launching...
Waiting for Darshan's iphone to report its views...
Debug service listening on ws://localhost:52975/ws
Syncing files to device Darshan's iphone...
```

pubspec.yaml:

```
dependencies:
  flutter:
    sdk: flutter
  firebase_core: ^0.5.0
  firebase_auth: ^0.18.0+1
  flutter_signin_button: ^1.0.0
  firebase_analytics: ^6.0.0
```

<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter: is fully installed. (Channel dev, 1.22.0-9.0.pre, on Mac OS X 10.15.2 19C57, locale
    en-US)
[!] Android toolchain - develop for Android devices: is partially installed; more components are
    available. (Android SDK version 29.0.3)
[✓] Xcode - develop for iOS and macOS: is fully installed. (Xcode 11.6)
[✓] Chrome - develop for the web: is fully installed.
[✓] Android Studio: is fully installed. (version 3.6)
[✓] Connected device: is fully installed. (5 available)
```

</details>

Can you provide your `pubspec.yaml` and also try to run following to see if it works ?

```
pod repo update
pod update
cd ../
flutter run
```

**Comment by jovanovski** [comment]
Same issue here, running on latest stable.

```
Analyzing dependencies
firebase_analytics: Using Firebase SDK version '6.26.0' defined in 'firebase_core'
firebase_core: Using Firebase SDK version '6.26.0' defined in 'firebase_core'
firebase_messaging: Using Firebase SDK version '6.26.0' defined in 'firebase_core'
[!] CocoaPods could not find compatible versions for pod "Firebase/Analytics":
  In Podfile:
    admob_flutter (from `.symlinks/plugins/admob_flutter/ios`) was resolved to 0.3.4, which depends on
      Firebase/Analytics

    firebase_analytics (from `.symlinks/plugins/firebase_analytics/ios`) was resolved to 6.0.0, which depends on
      Firebase/Analytics (~> 6.26.0)

CocoaPods could not find compatible versions for pod "Firebase/Core":
  In Podfile:
    firebase_analytics (from `.symlinks/plugins/firebase_analytics/ios`) was resolved to 6.0.0, which depends on
      Firebase/Analytics (~> 6.26.0) was resolved to 6.26.0, which depends on
        Firebase/Core (= 6.26.0)

    firebase_crashlytics (from `.symlinks/plugins/firebase_crashlytics/ios`) was resolved to 0.0.1, which depends on
      Firebase/Core

CocoaPods could not find compatible versions for pod "FirebaseAnalytics":
  In Podfile:
    admob_flutter (from `.symlinks/plugins/admob_flutter/ios`) was resolved to 0.3.4, which depends on
      FirebaseAnalytics (~> 6.1.3)

    firebase_analytics (from `.symlinks/plugins/firebase_analytics/ios`) was resolved to 6.0.0, which depends on
      Firebase/Analytics (~> 6.26.0) was resolved to 6.26.0, which depends on
        Firebase/Core (= 6.26.0) was resolved to 6.26.0, which depends on
          FirebaseAnalytics (= 6.6.0)

CocoaPods could not find compatible versions for pod "FirebaseCore":
  In Podfile:
    firebase_analytics (from `.symlinks/plugins/firebase_analytics/ios`) was resolved to 6.0.0, which depends on
      Firebase/Analytics (~> 6.26.0) was resolved to 6.26.0, which depends on
        Firebase/Core (= 6.26.0) was resolved to 6.26.0, which depends on
          FirebaseAnalytics (= 6.6.0) was resolved to 6.6.0, which depends on
            FirebaseCore (~> 6.7)

    firebase_analytics (from `.symlinks/plugins/firebase_analytics/ios`) was resolved to 6.0.0, which depends on
      Firebase/CoreOnly (~> 6.26.0) was resolved to 6.26.0, which depends on
        FirebaseCore (= 6.7.2)

    firebase_messaging (from `.symlinks/plugins/firebase_messaging/ios`) was resolved to 7.0.0, which depends on
      Firebase/Messaging (~> 6.26.0) was resolved to 6.26.0, which depends on
        FirebaseMessaging (~> 4.4.1) was resolved to 4.4.1, which depends on
          FirebaseCore (~> 6.6)
```

**Comment by AAverin** [comment]
Issue is resolved by
```
pod repo update
pod update
```
